sidestep_engine/models/gpu_utils.py

```
@@ -121,11 +121,17 @@ def detect_gpu(requested_device: str = "auto", requested_precision: str = "auto"
 def _cuda_free_mb(idx: int = 0) -> Optional[float]:
     """Return free CUDA memory in MiB, or None on failure."""
     try:
-        torch.cuda.synchronize(idx)
-        free, _total = torch.cuda.mem_get_info(idx)
-        return free / (1024 ** 2)
+        if torch.cuda.is_available():
+            torch.cuda.synchronize(idx)
+            free, _total = torch.cuda.mem_get_info(idx)
+            return free / (1024 ** 2)
+        elif torch.mps.is_available():
```

Ruff (0.15.7)

[warning] Unnecessary `elif` after `return` statement

Remove unnecessary `elif`

(RET505)